### ISSUE TITLE
Please add replay protection to prevent inadvertant sends on old chains

### ISSUE THREAD

**Issue by GriffGreen** [gandrewstone/bitcoin#383]
tl;DR
Replay protection will be implemented. It would be ideal if it were implemented client side as opposed to the user's side to protect the user's from losing their BTC. Trust me. The ETH/ETC fork was a shit show we should learn from.

-- 

The fact is Bitcoin is based on an opt-in mentality. Anyone who tries to enforce their opinion on others will fail, this is what I love most about this remarkable technology! No matter how small the minority, no one can be forced to adopt any new changes, everyone can run whatever software they prefer.

I worked for Slock.it and was very much an insider in the ETC/ETH hard fork. We made a conscious decision ignore replay protection out of arrogance. It was the wrong decision. 

By leaving it out we thought it would act as a deterrent, making it harder for ETC to exist, but this was a big mistake. There were people that felt strongly that the hard fork was unnecessary and they were able to maintain the value of their chain and over come the replay issues. Don't let history repeat itself.

It seems evident there is a fervent group of core supporters that will not support an unlimited hard forked chain no matter how much hash power you gather. 

With that information known, there needs to be a solution in place to prevent transactions from being replayed on both chains.  

When we hard forked, not including replay protection caused many people to lose value (e.g. exchanges where scammers (and non-scammers) were able to deposit [ETH] and then withdraw [ETH + ETC] continuously until all the exchange's ETC was drained) and IMHO we are at fault for being arrogant. 

Implementing replay protection will make it easy for your loyal users to dump the old Core bitcoins without losing the bitcoins on your chain.

**By not including replay protection, you impose upon every exchange and wallet service to implement a version of replay protection on their side.** 

They will need to protect themselves and their users, but there will be a % of these services that fail to implement the proper precautions causing a % of your loyal users to lose money and become disenchanted with digital currencies in general.

A hard fork requires people to update the software they are running, within that updated software, including a protection against replays is the responsible decision. It is a major step in the right direction to gain support for your fork.

When people feel too stupid to be their own bank we all lose... It's already hard enough to be into bitcoin if you are not technically savvy, please protect the adventurous non-tech user from losing their funds. Succeed where Ethereum failed.

**Comment by ftrader** [comment]
Appreciate the input, Griff. We are currently working on collecting information w.r.t. replay protection.

We are aware that there are several proposals around this, but it is also very important to get the "WHY" right as well as the "HOW".

To mention just one subtle point - the coming fork is not about BU only, it is about all implementations that want to transition to > 1MB blocks.
Implementing a change in transaction signatures (aka replay protection) would affect all other software that wants to maintain compatibility with the greatest-work chain.
We should ask what is the cost imposed on all those implementations too.

Another point that needs to be re-iterated is that replay 'protection' is something that is specific to a particular chain being mined. It is such a chain which wants to protect its transactions from being replayed by others. And it is the miners who validate these transactions before mining them into blocks. So they get to decide on what to 'protect' their chain against - not the BU project or implementation. BU can, at best, offer some tools to the users of its software, to allow them some choices in the matter. Such changes, which affect consensus, would be implemented by BUIPs, voted on by BU members. But anyone can submit technical proposals (even BUIPs) with specific ideas.
We may see such technical proposals if some decide they are technically worth proposing.

And the discussion about understanding transaction replay and its implications on the majority and minorities in a Bitcoin forking scenario is definitely open.

Let's explore it in depth, I see you raised a thread here:

https://bitco.in/forum/threads/bu-should-take-a-stance-against-anti-replay-measures.1786/#post-36403

**Comment by sickpig** [comment]
@GriffGreen 

BUcash edition added replay protection as per this specification.
